src/processor/operator/persistent/index_builder_error_handler.cpp

```diff
@@ -0,0 +1,61 @@
+#include "processor/operator/persistent/index_builder_error_handler.h"
+
+#include "processor/execution_context.h"
+#include "storage/store/node_table.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace processor {
+IndexBuilderErrorHandler::IndexBuilderErrorHandler(ExecutionContext* context,
+    common::LogicalTypeID pkType, storage::NodeTable* nodeTable, uint64_t queryID)
+    : ignoreErrors(context->clientContext->getClientConfig()->ignoreCopyErrors),
+      warningLimit(
+          std::min(context->clientContext->getClientConfig()->warningLimit, LOCAL_WARNING_LIMIT)),
+      context(context), pkType(pkType), nodeTable(nodeTable), queryID(queryID) {}
+
+void IndexBuilderErrorHandler::addNewVectors() {
+    offsetVector.push_back(std::make_shared<ValueVector>(LogicalTypeID::INTERNAL_ID,
+        context->clientContext->getMemoryManager()));
+    offsetVector.back()->state = DataChunkState::getSingleValueDataChunkState();
+    keyVector.push_back(
+        std::make_shared<ValueVector>(pkType, context->clientContext->getMemoryManager()));
+    keyVector.back()->state = DataChunkState::getSingleValueDataChunkState();
+}
```

Comment:
Made vectors size 1 because `NodeTable::delete_()` requires the vector to be size 1. If this assertion is safe to remove I can update to make vectors larger